File: src/main/java/com/example/librarymanagement/service/AuthorService.java
package com.example.librarymanagement.service;

import com.example.librarymanagement.model.entity.Author;

import java.util.List;

public interface AuthorService {
    Author createAuthor(Author author);
    Author updateAuthor(Long id, Author updatedAuthor);
    Author getAuthorById(Long id);
    List<Author> getAllAuthors();
}

File: src/main/java/com/example/librarymanagement/service/AuthorService.kt
package com.example.librarymanagement.service

import com.example.librarymanagement.model.domain.Author
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface AuthorService {
    fun createAuthor(author: Author): Mono<Author>
    fun updateAuthor(updatedAuthor: Author): Mono<Author>
    fun getAuthorById(id: String): Mono<Author>
    fun getAllAuthors(): Flux<Author>
}
